requests/maniphest_search.go

```diff
@@ -0,0 +1,116 @@
+		return nil
+	}
+
+	return json.Unmarshal(data, &o.Order)
+}
+
+// MarshalJSON creates JSON our of ManiphestRequestSearchOrder instance.
+func (o *ManiphestRequestSearchOrder) MarshalJSON() ([]byte, error) {
+	if o == nil {
+		return nil, errors.New("maniphest search order is nil")
+	}
+	if o.Builtin != "" {
+		return json.Marshal(o.Builtin)
+	}
+	if len(o.Order) > 0 {
+		return json.Marshal(o.Order)
+	}
+
+	return nil, nil
+}
+
+// SearchConstraints describes search criteria for request.
+type SearchConstraints struct {
+	// IDs - search for objects with specific IDs.
+	IDs            []string        `json:"ids,omitempty"`
+	// PHIDs - search for objects with specific PHIDs.
+	PHIDs          []string        `json:"phids,omitempty"`
+	// AssignedTo - search for tasks owned by a user from a list.
+	AssignedTo     []string        `json:"assigned,omitempty"`
+	// Authors - search for tasks with given authors.
+	Authors        []string        `json:"authorPHIDs,omitempty"`
+	// Statuses - search for tasks with given statuses.
+	Statuses       []string        `json:"statuses,omitempty"`
+	// Priorities - search for tasks with given priorities.
+	Priorities     []string        `json:"priorities,omitempty"`
+	// Subtypes - search for tasks with given subtypes.
+	Subtypes       []string        `json:"subtypes,omitempty"`
+	// Column PHIDs ??? - no doc on phab site.
+	ColumnPHIDs    []string        `json:"columnPHIDs,omitempty"`
+	// OpenParents - search for tasks that have parents in open state.
+	OpenParents    *bool           `json:"hasParents,omitempty"`
+	// OpenSubtasks - search for tasks that have child tasks in open state.
+	OpenSubtasks   *bool           `json:"hasSubtasks,omitempty"`
+	// ParentIDs - search for children of these parents.
+	ParentIDs      []int           `json:"parentIDs,omitempty"`
+	// SubtaskIDs - Search for tasks that have these children.
+	SubtaskIDs     []int           `json:"subtaskIDs,omitempty"`
+	// CreatedAfter - search for tasks created after given date.
+	CreatedAfter   *entities.Epoch `json:"createdStart,omitempty"`
+	// CreatedBefore - search for tasks created before given date.
+	CreatedBefore  *entities.Epoch `json:"createdEnd,omitempty"`
+	// ModifiedAfter - search for tasks modified after given date.
+	ModifiedAfter  *entities.Epoch `json:"modifiedStart,omitempty"`
+	// ModifiedBefore - search for tasks modified before given date.
+	ModifiedBefore *entities.Epoch `json:"modifiedEnd,omitempty"`
+	// ClosedAfter - search for tasks closed after given date.
+	ClosedAfter    *entities.Epoch `json:"closedStart,omitempty"`
+	// ClosedBefore - search for tasks closed before given date.
+	ClosedBefore   *entities.Epoch `json:"closedEnd,omitempty"`
+	// ClosedBy - search for tasks closed by people with given PHIDs.
+	ClosedBy       []string        `json:"closerPHIDs,omitempty"`
+	// Query - find objects matching a fulltext search query. See "Search User Guide" in the documentation for details.
```

Comment:
The "See user guide" section isn't needed here. Probably a bad copy-paste.

```suggestion
	// Query - find objects matching a fulltext search query.
```